### ISSUE TITLE
IDE0004 Unnecessary cast false positive

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#58804]
**Version Used**: 

main

**Steps to Reproduce**:

```csharp
class C
{
    static object M(object o)
    {
        return (object)o.ToString;
    }
}
```

**Expected Behavior**:

No unnecessary cast is reported.

**Actual Behavior**:

Unnecessary cast is reported. Removing the cast will produce this warning:

> warning CS8974: Converting method group 'ToString' to non-delegate type 'object'. Did you intend to invoke the method?

Extracted from https://github.com/dotnet/roslyn/issues/50043#issuecomment-924209443
